src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -881,11 +872,22 @@ struct MutationContext
     bool need_prefix = true;
 
     scope_guard temporary_directory_lock;
+
+    bool checkOperationIsNotCanceled() const
+    {
+        if (merges_blocker->isCancelledForPartition(new_data_part->info.partition_id)
```

Comment:
I assume here that the merge operation doesn't create parts in other partitions, so we might check only if the destination partition was canceled or not.